crates/rpc/rpc-eth-api/src/helpers/trace.rs

```diff
@@ -196,8 +196,11 @@ pub trait Trace: LoadState {
                     tx.hash,
                 )?;
 
-                let env =
-                    EnvWithHandlerCfg::new_with_cfg_env(cfg, block_env, tx_env_with_recovered(&tx));
+                let env = EnvWithHandlerCfg::new_with_cfg_env(
+                    cfg,
+                    block_env,
+                    Call::evm_config(&this).tx_env(&tx),
+                );
```

Comment:
me thinks we want a trait function for this as part of EthTransactions, but can do as followup